code/game/machinery/mind_machine.dm

```diff
@@ -0,0 +1,846 @@
+			swapOccupants()
+			unlockPods()
+		else
+			errorMessage = "Missing occupant"
+
+/obj/machinery/mind_machine/mind_machine_hub/proc/scanOccupants(var/mob/S)
+	if((!locate(occupantOne,connectOne.contents)) || (!locate(occupantTwo, connectTwo.contents)))
+		errorMessage = "Missing occupant"
+		occupantOne = null
+		occupantTwo = null
+		return
+	currentlySwapping = TRUE
+	lockedPods = TRUE
+	if((connectOne.icon_state != "mind_pod_closed") && (connectTwo.icon_state != "mind_pod_closed"))
+		playsound(connectOne, 'sound/machines/poddoor.ogg', 55, 1)
+		playsound(connectTwo, 'sound/machines/poddoor.ogg', 55, 1)
+		flick("mind_pod_closing",connectOne)
+		connectOne.icon_state = "mind_pod_closed"
+		flick("mind_pod_closing",connectTwo)
+		connectTwo.icon_state = "mind_pod_closed"
+	sleep(2 SECONDS)
+	playsound(connectOne, 'sound/effects/sparks4.ogg', 80, 1)
+	playsound(connectTwo, 'sound/effects/sparks4.ogg', 80, 1)
+	occupantScan = TRUE
+	S = occupantOne
+	occupantNameOne = S.name
+	switch(S.stat)
+		if(0)
+			occupantStatOne = "Alive"
+		if(1)
+			occupantStatOne = "Unconscious"
+		if(2)
+			occupantStatOne = "Dead"
+	if(!S.mind)
+		mindTypeOne = "Lower" //Simple mob
+	if(S.mind)
+		if(isrobot(S))
+			mindTypeOne = "Artificial" //Silicon player, obviously
+		else
+			mindTypeOne = "Higher" //Player controlled
+	if(isvampire(S) || isanycultist(S) || ischangeling(S) || ismalf(S))
+		mindTypeOne = "Shielded" //Mostly to fix spell bugs but also tinfoil
+	if((ishigherbeing(S)) || (ismonkey(S)))
+		var/mob/living/carbon/T = S
+		if(T.is_wearing_item(/obj/item/clothing/head/tinfoil))
+			mindTypeTwo = "Shielded"
+	S = occupantTwo
+	occupantNameTwo = S.name //and for pod two
+	switch(S.stat)
+		if(0)
+			occupantStatTwo = "Alive"
+		if(1)
+			occupantStatTwo = "Unconscious"
+		if(2)
+			occupantStatTwo = "Dead"
+	if(!S.mind)
+		mindTypeTwo = "Lower"
+	if(S.mind)
+		if(isrobot(S))
+			mindTypeTwo = "Artificial"
+		else
+			mindTypeTwo = "Higher"
+	if(isvampire(S) || isanycultist(S) || ischangeling(S) || ismalf(S))
+		mindTypeTwo = "Shielded"
+	if((ishigherbeing(S)) || (ismonkey(S)))
+		var/mob/living/carbon/T = S
+		if(T.is_wearing_item(/obj/item/clothing/head/tinfoil))
+			mindTypeTwo = "Shielded"
+	currentlySwapping = FALSE
+
+/obj/machinery/mind_machine/mind_machine_hub/proc/swapOccupants(var/mob/living/M)
+	if(occupantScan != TRUE || lockedPods != TRUE || badSwap == TRUE)
+		return
+	if((!locate(occupantOne,connectOne.contents)) || (!locate(occupantTwo, connectTwo.contents)))
+		errorMessage = "Occupants missing" //sanity and wizards exist
+		occupantOne = null
+		occupantTwo = null
+		unlockPods()
+		return
+	if(bluespaceConduit == 0)
+		errorMessage = "Conduit requires charge"
+		return
+	if(mindTypeOne == "Shielded" || mindTypeTwo == "Shielded")
+		errorMessage = "Critical error, aborting"
+		spark(src)
+		spark(connectOne)
+		spark(connectTwo)
+		unlockPods()
+		return
+	if(occupantStatOne == "Dead" || occupantStatTwo == "Dead")
```

Comment:
Here's a fun problem: what if the mob dies while *inside* the pod. Then this variable won't have updated correctly because it's pulled only in the scan proc.

Reply:
That was actually intentional. Since it already scanned the mind I figured it would make sense to be able to transfer it out. It would also work as a sort of ghetto DNA disk as long as the pod wasn't touched in the meantime.

Reply:
May want to leave some sort of comment then to say "yes, this intentional".